lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -44,18 +44,21 @@ local function playerIsInBuild( ply )
     return !getPlayerPvpMode( ply )
 end
 
-local function movementMultiplier(weaponCount) 
+local function movementMultiplier(weaponCount)
+    if weaponCount < 1 then return 1 end
     return math.Clamp(1 - (1.9^( weaponCount))/100, 0, 1)
 end
 
 local function setSpeed(ply, multiplier) 
     -- set speed multiplier, 0 - 1
-    ply:SetRunSpeed(math.Clamp(baseRunSpeed*multiplier,minRunSpeed, baseRunSpeed))
-    ply:SetWalkSpeed(math.Clamp(baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed))
+    local runSpeed = baseRunSpeed * multiplier
+    local walkSpeed = baseWalkSpeed * multiplier
+    ply:SetRunSpeed( math.Clamp( runSpeed, minRunSpeed, baseRunSpeed ) )
+    ply:SetWalkSpeed( math.Clamp( walkSpeed, minWalkSpeed, baseWalkSpeed ) )
```

Comment:
I think using clamp here is a bad idea. Our math should simply return the proper value.

Clamping like this prevents us from increasing the player speed if they only have a melee weapon, which is something we were looking at doing at one point.

Reply:
Yeah `.max` seems like a good choice.